docs/core/howto/logger.rst

```diff
@@ -277,10 +277,33 @@ System-provided event keys include:
 Avoid mutable event keys
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
-Emitting applications should be cautious about inserting objects into event which may be mutated later.
+Emitting applications should be cautious about inserting objects into events which may be mutated later.
 While observers are called synchronously, it is possible that an observer will do something like queue up the event for later serialization, in which case the serialized object may be different than intended.
 
 
+Capturing log events for testing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you want to test that your code is logging the expected events, you can use the :api:`twisted.logger.LogCapture <LogCapture>` context manager:
+
+.. code-block:: python
+
+    from twisted.logger import Logger, LogCapture, LogLevel
+    from twisted.trial.unittest import TestCase
+
+    class SomeTests(TestCase):
+
+        def test_capture(self):
+            with LogCapture() as capture:
```

Comment:
2 minor nitpicks here:

Does this really need to be its own whole object?  My inclination would be to just use a `list`; less API surface to care about supporting.  If we really want `asText`, then  `"".join(map(eventAsText, events))` could go in a public utility function.

In the various places I've had to implement this over the years, I've typically spelled it:

```
with capturedLogs() as captured:
   ...
# captured is a list of all the events now
```
and I feel like this reads a little better in english.

Reply:
@adiroiban Any thoughts about keeping `asText()`?  I'm inclined to agree with Glyph that it's not needed.  It can also be a function we provide later if needed.

Reply:
@glyph: [this](https://github.com/burningmantech/ranger-deploy/blob/master/src/deploy/ext/logger.py#L41) is what I had originally written in my app, which is I think is more like what you have in mind.

Reply:
The simple list looks good.

I guess that later we can return an instance which inherits from list and has an extra `.asText()` method :)

Thanks for the change :)